### PR TITLE
resolve simultaneous port usage for #522

### PR DESCRIPTION
**Pull request by ixje**
**What current issue(s) does this address, or what feature is it adding?**
Resolve simultaneous connecting from and to the same port. Part of addressing https://github.com/CityOfZion/neo-python/issues/522

**How did you solve this problem?**
changed from using the `endpoint` API to lower level `reactor` API. While the original code works in standalone mode, it seems to conflict when used in conjunction with `connectTCP`
https://github.com/CityOfZion/neo-python/blob/f003f0a66cdd10cef91f30a6cda5549844ebd013/neo/Network/NodeLeader.py#L151

**How did you make sure your solution works?**
manual by running the node, then typing `nodes` to verify that we connected to our seeds, then run `nc localhost 20333` and see a `version` reply.
```
Eriks-Air:~ erik$ nc 127.0.0.1 20333
Anttversion2
�7��[mO��x�/NEO-PYTHON:0.7.8-dev/
```

**Are there any special changes in the code that we should be aware of?**
no

**Please check the following, if applicable:**

- [ ] Did you add any tests?
- [X] Did you run `make lint`?
- [X] Did you run `make test`?
- [X] Are you making a PR to a feature branch or development rather than master?
- [X] Did you add an entry to `CHANGELOG.rst`? (if not, please do)

no new changelog entry, just polishing old PR

### CODE REVIEW

File: neo/Network/test_node_leader.py [hunk 1]
```diff
@@ -141,6 +139,7 @@ def mock_send_msg(node, message):
         with patch('twisted.internet.reactor.connectTCP', mock_connect_tcp):
             with patch('twisted.internet.reactor.callLater', mock_call_later):
                 with patch('neo.Network.NeoNode.NeoNode.SendSerializedMessage', mock_send_msg):
+                    leader.Shutdown()
```
Comment (ixje) [inline neo/Network/test_node_leader.py]: this was the only place that seemed to work. I tried placing it in the `teardown` and just as final statement for tests using the leader instance, but no luck. Don't know why, but this worked

**Comment by ixje** [comment]
ugh, it builds fine locally just not with the `pexpect` tests on travis. probably has something to do with timeouts. Will try to restart the build somewhere early in the morning when there's hopefully more resources available or something like that :/

**Comment by jseagrave21** [comment]
@ixje Hey, I when I reduced the timeout from their default, I was just trying to speed up the test. You can always remove the `timeout=...` and it should default to 30 sec. Then the test should pass.

**Comment by localhuman** [comment]
I think it makes sense to turn it off by default.

**Comment by ixje** [comment]
@localhuman we're good now 💦

**Comment by jseagrave21** [comment]
@ixje With these changes, will P2P still function, if I turn it on? Also, is there any additional benefits (to the node, to the network) by turning on P2P?

**Pull request merged**